File: app/src/main/java/cz/kutner/comicsdb/activity/SearchActivity.java
package cz.kutner.comicsdb.activity;

import android.app.SearchManager;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;

import com.crashlytics.android.answers.Answers;
import com.crashlytics.android.answers.SearchEvent;
import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;

import butterknife.Bind;
import butterknife.ButterKnife;
import cz.kutner.comicsdb.ComicsDBApplication;
import cz.kutner.comicsdb.R;
import cz.kutner.comicsdb.adapter.SearchPagerAdapter;

public class SearchActivity extends AppCompatActivity {
    @Bind(R.id.pager)
    ViewPager pager;
    @Bind(R.id.sliding_tabs)
    TabLayout slidingTabs;
    FragmentPagerAdapter fragmentPagerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);
        ButterKnife.bind(this);
        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        fragmentPagerAdapter = new SearchPagerAdapter(getSupportFragmentManager(), getIntent());
        pager.setAdapter(fragmentPagerAdapter);
        slidingTabs.setupWithViewPager(pager);
        String query = getIntent().getStringExtra(SearchManager.QUERY);
        Answers.getInstance().logSearch(new SearchEvent()
                .putQuery(query));
        Tracker tracker = ComicsDBApplication.getTracker();
        tracker.send(new HitBuilders.EventBuilder().setCategory("Search").setAction(query).build());
    }
}

File: app/src/main/kotlin/cz/kutner/comicsdb/activity/SearchActivity.kt
package cz.kutner.comicsdb.activity

import android.app.SearchManager
import android.app.inject
import android.os.Bundle
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v7.app.AppCompatActivity
import com.google.firebase.analytics.FirebaseAnalytics
import cz.kutner.comicsdb.R
import cz.kutner.comicsdb.adapter.SearchPagerAdapter
import kotlinx.android.synthetic.main.activity_tabbed.*

class SearchActivity : AppCompatActivity() {
    val firebase by inject<FirebaseAnalytics>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tabbed)
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
        val fragmentPagerAdapter: FragmentStatePagerAdapter = SearchPagerAdapter(supportFragmentManager, intent, applicationContext)
        pager.adapter = fragmentPagerAdapter
        sliding_tabs.setupWithViewPager(pager)
        val query = intent.getStringExtra(SearchManager.QUERY)

        val bundle: Bundle = Bundle()
        bundle.putString(FirebaseAnalytics.Param.SEARCH_TERM, query)
        firebase.logEvent(FirebaseAnalytics.Event.SEARCH, bundle)

    }
}
